### PR TITLE
all: replace usage of `java.util.Date` and `java.time.LocalDateTime` with `java.time.Instant`

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Replace usage of java.util.Date and java.time.LocalDateTime with `java.time.Instant`. Also, updated the gson serializer to make sure `java.time.Instant` is serialized/deserialized using a readable format in the UTC timezone, like `"2022-03-01T01:01:01.544Z"` for instance.

### Why

We were using different objects to represent dates, which was pretty inconsistent.

Additionally, Gson does not serialize java.time.* objects to a string (by default) so we had to use a custom Gson instance to make sure we would serialize things accordingly.

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/util/GsonUtils.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import java.time.Instant;
+
+public class GsonUtils {
+  private static Gson instance = null;
+
+  public static GsonBuilder builder() {
+    return new GsonBuilder().registerTypeAdapter(Instant.class, new InstantConverter());
+  }
+
+  public static Gson getGsonInstance() {
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/util/GsonUtils.java:14]: In Java Singleton pattern, this is usually called `getInstance()`
https://www.baeldung.com/java-singleton

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java [hunk 2]
```diff
@@ -12,9 +12,11 @@
 @RestController
 public class RateController {
   private final RateService rateService;
+  private final Gson gson;
 
-  public RateController(AppSettings appSettings, RateService rateService) {
+  public RateController(AppSettings appSettings, RateService rateService, Gson gson) {
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java:17]: I am not sure if it is too much to autowired it? 😄  

Should we use GsonUtils.builder().create()?

**Review by lijamie98 — APPROVED** [review]
LGTM. 

I need this PR fo fix my `java.time.Instant` serialization problem. 😄

File: core/src/main/java/org/stellar/anchor/util/GsonUtils.java (hunk 1 above)
Comment (marcelosalloum) [reply core/src/main/java/org/stellar/anchor/util/GsonUtils.java]: https://github.com/stellar/java-stellar-anchor-sdk/pull/158/commits/130707dffdc2cc60462b7b10a8f0bc33f73fb739

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java (hunk 2 above)
Comment (marcelosalloum) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java]: https://github.com/stellar/java-stellar-anchor-sdk/pull/158/commits/923fdb5b54648c99affd3eda00fd458c792a217f

**Pull request merged**